californium-core/src/main/java/org/eclipse/californium/core/network/UdpMatcher.java

```diff
@@ -396,11 +398,12 @@ public void completed(final Exchange exchange) {
 
 		@Override
 		public void contextEstablished(final Exchange exchange) {
-
-			if (exchange.getRequest() != null) {
-				observationStore.setContext(exchange.getRequest().getToken(), exchange.getCorrelationContext());
+			Request currentRequest = exchange.getCurrentRequest(); 
+			Request request = exchange.getRequest(); 
+			if (request == currentRequest && request != null && request.isObserve()) {
+				observationStore.setContext(request.getToken(), exchange.getCorrelationContext());
```

Comment:
I agree with @sbernard31 and I am sorry, but I am not able to understand the point of your latest comment, @boaks. Can you please elaborate/rephrase what your intention is?